orc/src/main/java/org/apache/iceberg/orc/OrcToIcebergVisitorWithPseudoId.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.orc;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.types.Types.NestedField;
+import org.apache.orc.TypeDescription;
+
+public class OrcToIcebergVisitorWithPseudoId extends OrcToIcebergVisitor {
+
+  private static final String PSEUDO_ICEBERG_FIELD_ID = "-1";
+
+  @Override
+  public Optional<NestedField> union(TypeDescription union, List<Optional<NestedField>> options) {
+    union.setAttribute(org.apache.iceberg.orc.ORCSchemaUtil.ICEBERG_ID_ATTRIBUTE, PSEUDO_ICEBERG_FIELD_ID);
+    List<Optional<NestedField>> optionsCopy = new ArrayList<>(options);
```

Comment:
nit:
more efficient to create an immutable copy of the list directly
```suggestion
    List<Optional<NestedField>> optionsCopy = List.copyOf(options);
```

Reply:
Are you saying the guava api? guava doesn't have this api i just check, it only has `Lists.newArrayList()` . which is the same as `new arraylist` i think.

Reply:
and, in fact, the next line, i need to insert an element at the start of the List, so it cannot be immutable.